### PR TITLE
[Feature Fix] Fix modals overlap with growlbox error messages

### LINKED ISSUE #1340: Modals overlap with growlbox error messages
Steps
--------
On develop, in the code for adding a link to a large project, change the submit function so that it fails. I did this by changing this URL to something that does not exist: https://github.com/CenterForOpenScience/osf.io/blob/develop/website/static/js/pointers.js#L95

Expected
------------
I expected the modal to go away, showing the growlbox error message. 

Actual
--------
The growlbox error message shows up while the modal is still present, making it difficult to see: 
screen shot 2014-11-25 at 5 54 10 pm

Possible Solutions
-------------
- Dismiss the modal when an error occurs so that the growlbox can be seen 
- Show the error message in the modal 

All modals on the OSF should be checked for this behavior.

### PR DESCRIPTION
**Pull request by haoyuchen1992**
### Purpose
This PR is to solve the issue that when growlbox error message is sent, it will be hidden by modal. 

**Lesson:** 
when `handleJSONError` from `osfhelper` and modal are used at the same time,  if the modal is not from `bootbox`, we must hidden the modal first. Because  growlbox has been used inside `handleJSONError`. When modal isn't hidden, the error message (growlbox) can't be seem. 

Resolve #1340 

### Changes
I checked all the use cases of `handleJSONError` and found there were two places that had the reported issue with the same pattern. They are in the functions --`fetchResults` and `submit` in `AddPointerViewModel`.
1) Create modal DOM variable
2) Set the modal DOM as 'hidden' before invoking `handleJSONError`

### Side Effects
None.

### CODE REVIEW

**Comment by lyndsysimon** [comment]
Looks great to me :)

File: website/static/js/pointers.js [hunk 1]
```diff
@@ -63,9 +64,10 @@ var AddPointerViewModel = oop.extend(Paginator, {
                 self.currentPage(result.page);
                 self.numberOfPages(result.pages);
                 self.addNewPaginators();
-            }).fail(
-                osfHelpers.handleJSONError
-            );
+            }).fail(function() {
+                self.elm.modal('hide');
+                osfHelpers.handleJSONError();
```
Comment (sloria) [inline website/static/js/pointers.js]: `handleJSONError` takes the `xhr` as its first argument. You should pass the `xhr` from the callback to the `handleJSONError` function.

```js
.fail(function(xhr) {
    osfHelpers.handleJSONError(xhr);
}
```

File: website/static/js/pointers.js [hunk 2]
```diff
@@ -175,7 +178,7 @@ function PointerManager(selector, nodeName) {
     self.selector = selector;
     self.$element = $(self.selector);
     self.nodeName = nodeName;
-    self.viewModel = new AddPointerViewModel(nodeName);
+    self.viewModel = new AddPointerViewModel({nodeTitle:nodeName, elm:self.$element});
```
Comment (sloria) [inline website/static/js/pointers.js]: Add space after the `:`.

**Comment by sloria** [comment]
While this PR works well, it would be an improvement to show the error message in the modal itself, just below the two tables. 

In general, the growlbox shown by `handleJSONError` should only be used in the event of an *unexpected* error, e.g. 500 errors. Validation errors and other expected errors should be displayed near their inputs, as per our style guidelines.

@haoyuchen1992 Can you look into this? If it's an easy change, go ahead and make it. If not, then we can defer it for now.

**Comment by haoyuchen1992** [comment]
@sloria I tried to show the error message by using bootstrap alert. Here is the result --
When search doesn't work:
screen shot 2015-06-29 at 4 19 13 pm

When submitting has error:
screen shot 2015-06-29 at 4 20 00 pm

**Comment by sloria** [comment]
That UI is inconsistent with the rest of the site; we typically use just red text (`text-danger` within a `.help-block`) *above* the submit button.

**Comment by sloria** [comment]
Also, try to avoid manual DOM updates where possible; prefer to bind observables to the HTML.

**Comment by sloria** [comment]
Pass finished. :v:

**Comment by haoyuchen1992** [comment]
@sloria  Following the design routine. Here is the result.

When searching has error:
screen shot 2015-06-30 at 10 20 07 am

When submitting has error:
screen shot 2015-06-30 at 10 19 32 am

**Comment by sloria** [comment]
@haoyuchen1992 How did you generate the errors above?

**Comment by haoyuchen1992** [comment]
@sloria I changed URL  into a wrong one. For example, Change `url: nodeApiUrl + 'pointer/'` into `url: nodeApiUrl + 'pointer-----/'`

**Comment by sloria** [comment]
Thanks for the clarification.

**Pull request merged**